src/Microsoft.DotNet.ApiSymbolExtensions/SymbolExtensions.cs

```diff
@@ -91,6 +94,7 @@ symbol.DeclaredAccessibility switch
                 Accessibility.Protected => includeEffectivelyPrivateSymbols || symbol.ContainingType == null || !IsEffectivelySealed(symbol.ContainingType, includeInternals),
                 Accessibility.ProtectedOrInternal => includeEffectivelyPrivateSymbols || includeInternals || symbol.ContainingType == null || !IsEffectivelySealed(symbol.ContainingType, includeInternals),
                 Accessibility.ProtectedAndInternal => includeInternals && (includeEffectivelyPrivateSymbols || symbol.ContainingType == null || !IsEffectivelySealed(symbol.ContainingType, includeInternals)),
+                Accessibility.Private => includeEffectivelyPrivateSymbols && IsExplicitInterfaceImplementation(symbol),
```

Comment:
I don't understand that change. Can you please elaborate? From what I see here, it looks wrong as private API isn't the same as an effectively private symbol (which is protected).

Reply:
```C#
using System;

public class MemoryManager : System.IDisposable
{
    /// you can call Dispose() only through pointer of `IDisposable` and not through pointer to `MemoryManager`
    void System.IDisposable.Dispose() { Console.WriteLine("1"); }
}

public class Test
{
	static void CallThroughIDisposableDirect(System.IDisposable mm)
	{
		mm.Dispose();
	}

	public static void Main()
	{
		MemoryManager mm = new MemoryManager();
		CallThroughIDisposableDirect(mm);
        // mm.Dispose(); <-- failure
	}
}
```

ExplicitInterfaceImplementation methods have `Accessibility.Private` in concrete example above.
Before, we filter out all private methods in `_ => includeInternals && symbol.DeclaredAccessibility != Accessibility.Private,`. if `includeEffectivelyPrivateSymbols` is false - we must (as I see this) provide just public interface..

Reply:
I forgot to mention, the the SyntaxGenerator does not support generation of ExplicitInterfaceImplementation methods and that's why I added support in the GenAPI. in concreate example above, SyntaxGenerator is generating
```C#
public class MemoryManager : System.IDisposable
{
     void System.IDisposable.Dispose() { Console.WriteLine("1"); }
}
```
-->
```
public class MemoryManager : System.IDisposable
{
}
```
as a result, compilation error.

Reply:
Ok. I'll add new parameter `bool includeExplicitInterfaceImplementationSymbols = false`